scripts/lib/api/processHtml.test.ts

```diff
@@ -0,0 +1,300 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import { CheerioAPI, Cheerio, load as cheerioLoad } from "cheerio";
+import { describe, expect, test } from "@jest/globals";
+
+import {
+  addLanguageClassToCodeBlocks,
+  loadImages,
+  handleSphinxDesignCards,
+  maybeSetModuleMetadata,
+  renameAllH1s,
+  removeHtmlExtensionsInRelativeLinks,
+  removeDownloadSourceCode,
+  removePermalinks,
+  removeColonSpans,
+  replaceSourceLinksWithGitHub,
+} from "./processHtml";
+import { Metadata } from "./Metadata";
+
+class Doc {
+  readonly $: CheerioAPI;
+  readonly $main: Cheerio<any>;
+
+  constructor($: CheerioAPI, $main: Cheerio<any>) {
+    this.$ = $;
+    this.$main = $main;
+  }
+
+  static load(html: string): Doc {
+    const $ = cheerioLoad(`<div role="main">${html}</div>`);
+    return new Doc($, $("[role='main']"));
+  }
+
+  html(): string {
+    return this.$main.html()!.trim();
+  }
+
+  expectHtml(expected: string): void {
+    expect(this.html()).toEqual(expected.trim());
+  }
+}
+
+describe("loadImages()", () => {
+  test("normal file", () => {
+    const doc = Doc.load(
+      `<img src="../_static/logo.png" alt="Logo"><img src="../_static/images/view-page-source-icon.svg">`,
+    );
+    const images = loadImages(
+      doc.$,
+      doc.$main,
+      "http://localhost:3000/api/my-file.html",
+      "/my-images",
+      false,
+    );
+    expect(images).toEqual([
+      {
+        src: "http://localhost:3000/_static/logo.png",
+        dest: "/my-images/logo.png",
+      },
+      {
+        src: "http://localhost:3000/_static/images/view-page-source-icon.svg",
+        dest: "/my-images/view-page-source-icon.svg",
+      },
+    ]);
+    doc.expectHtml(
+      `<img src="/my-images/logo.png" alt="Logo"><img src="/my-images/view-page-source-icon.svg">`,
+    );
+  });
+
+  test("release note", () => {
+    const doc = Doc.load(
+      `<img src="../_static/images/view-page-source-icon.svg">`,
+    );
+    const images = loadImages(
+      doc.$,
+      doc.$main,
+      "http://localhost:3000/api/release_notes.html",
+      "/my-images/0.45",
+      true,
+    );
+    expect(images).toEqual([
+      {
+        src: "http://localhost:3000/_static/images/view-page-source-icon.svg",
+        dest: "/my-images/view-page-source-icon.svg",
+      },
+    ]);
+    doc.expectHtml(`<img src="/my-images/view-page-source-icon.svg">`);
+  });
+});
+
+test("handleSphinxDesignCards()", () => {
```

Comment:
This test has a lot of HTML and it's hard to see what's happening. Idk if I should add comments to the test? Or is it reasonable that people look at the source code to see what's happening? For example, the source code already explains what it does in the docstring.